pystac_client/item_search.py

```diff
@@ -145,6 +147,7 @@ class ItemSearch:
         method: The http method, 'GET' or 'POST'
         stac_io: An instance of of StacIO for retrieving results. Normally comes from the Client that returns this ItemSearch
         client: An instance of a root Client used to set the root on resulting Items
+        q: Freetext string as per the STAC API `free-text-search` extension
```

Comment:
Since this is a third-party extension, it seems appropriate to add a link to the [extension definition](https://github.com/cedadev/stac-freetext-search) in the docs (IMO).